rust/lance-encoding/src/encoder.rs

```diff
@@ -476,8 +484,10 @@ impl ArrayEncodingStrategy for CoreArrayEncodingStrategy {
         let use_dict_encoding = data_type == &DataType::Utf8
             && check_dict_encoding(arrays, get_dict_encoding_threshold());
 
+        let stats = arrays.compute_statistics();
```

Comment:
I agree that `create_array_encoder` is the correct place to calculate the statistics today.